cogs/music.py

```diff
@@ -0,0 +1,180 @@
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+import disnake
+from disnake.ext.commands import command
+
+from utils.distyping import Context
+from service.music import MasterPlayer
+from .commandbase import CommandBase
+
+if TYPE_CHECKING:
+    from utils.distyping import ManChanBot
+
+class Music(CommandBase):
+    def __init__(self, bot: ManChanBot):
+        self.bot = bot
+        self.master_player = MasterPlayer()
+        self.Embed = disnake.Embed(title="MANCHAN RADIO")
+
+    @command()
+    async def setup_music(self, ctx: Context):
+        guild_id = ctx.guild.id
+        if (not guild_id):
+             return await ctx.channel.send("Please restart", delete_after= 5)
+        if (self.master_player.getPlayer(guild_id) and 
+            self.master_player.getPlayer(guild_id).channel_id and
+            self.master_player.getPlayer(guild_id).player_ui):
+            return await ctx.channel.send("channel is made", delete_after= 5)
+        channel = await ctx.guild.create_text_channel(name="manchan radio")
+        player_ui = await channel.send(embed=self.Embed)
+        self.master_player.createPlayer(guild_id)
+        self.master_player.getPlayer(guild_id).set_channel_id(channel.id)
+        self.master_player.getPlayer(guild_id).set_player_ui(player_ui)
+        await ctx.message.delete()
+
+    @command()
+    async def play(self, ctx: Context, *args: str):
+        guild_id = ctx.guild.id
+        player = self.master_player.getPlayer(guild_id)
+        if not player.get_channel_id() == ctx.channel.id:
+            return
+
+        if player.is_paused:
+            player.is_paused = False
```

Comment:
No, because there is not playing because there is no song and not playing because the audio is paused, without the check if you do this command it will skip the current paused song not resume it, which is not the desired behavior

Reply:
Wouldnt that not be an issue then if this command has different functionality if a string argument is given?

Scenarios:
1. Bot is not in vc. I do `!play`. Bot joins but with no song playing, `is_playing = false`
2. Bot is not in vc. I do `!play vamp anthem`. Bot joins and starts playing the song. `is_playing = true`
3. Bot is in vc and `is_playing = true`. I do `!play tokyo shoegaze`. Bot will then skip to next song. `is_playing = true`
4. Bot is in vc and `is_playing = true`. I do `!play`. Bot does nothing or yells at me. `is_playing = true`
5. Bot is in vc and `is_playing = false`. I do `!play duvet boa`. Bot will start playing the song. `is_playing = true`
6. Bot is in vc and `is_playing = false`. I do `!play`. Bot gets angry. `is_playing = false`


So essentially TLDR: if song name given, start playing that song. If no song name given, attempt to join the vc but if it is already there then send an error

Reply:
Or actually it can be queued instead of skipped over when !play <song>.


I think instead this command could be split into join() and resume() or play() (resume or play can be aliased so both work).

And then in a future ticket a more sophisticated command can be made that can do both or handle those scenarios i listed

Reply:
addressed, !play now can do `!play tokyo shoegaze` and can unpause the audio with `!play`